src/zap.ts

```diff
@@ -0,0 +1,44 @@
+import { ChainId, isEthereum, isFantom } from "./chain";
+import { Token, VaultMetadataOverrides } from "./types";
+
+type ZapInOptions = { isZapInSupported: boolean; zapInWith?: keyof Token["supported"] };
+
+type ZapInProps = {
+  chainId: ChainId;
+  token?: Partial<Token>;
+  vaultMetadata: VaultMetadataOverrides | null;
+};
+
+export function getZapInOptions({ chainId, token, vaultMetadata }: ZapInProps): ZapInOptions {
+  if (!token?.supported || !vaultMetadata) {
+    return { isZapInSupported: false };
+  }
+
+  if (isEthereum(chainId)) {
+    const { zapInWith } = vaultMetadata;
+
+    if (zapInWith && isValidZap(zapInWith, token.supported)) {
+      const isZapInSupported = !!token.supported[zapInWith];
+      return { isZapInSupported, zapInWith };
+    }
+  }
+
+  if (isFantom(chainId)) {
+    const zapInWith = "ftmApeZap"; // Hardcoded for now
```

Comment:
Ideally this would come from the `vaultMetadata` and we then can ignore the chain and just do https://github.com/yearn/yearn-sdk/pull/280/files#diff-3f1abc3b62c51b60a677aa685112987c1b861d2c476c8e45d41bbaffedaa5155R18-R23 for all chains